src/components/VmDetails/cards/DetailsCard/index.js

```diff
@@ -253,6 +278,13 @@ class DetailsCard extends React.Component {
     if (prevProps.templates !== this.props.templates) {
       this.setState({ templateList: createTemplateList(this.props.templates) }) // eslint-disable-line react/no-did-update-set-state
     }
+
+    if (prevProps.operatingSystems !== this.props.operatingSystems) {
+      const clusterId = this.props.vm.getIn(['cluster', 'id'])
+      const cluster = this.props.clusters && this.props.clusters.get(clusterId)
+      const architecture = cluster && cluster.get('architecture')
+      this.setState({ osList: createOsList(architecture) }) // eslint-disable-line react/no-did-update-set-state
```

Comment:
I suggest pushung the `clusterId`, `cluster` and `architecture` lookups to `createOsList()` and just pass in the VM and Clusters.  That way those lookups only happen in `createOsList()` instead of being repeated here and in the constructor.